addons/l10n_nl/models/account_journal.py

```diff
@@ -8,16 +8,18 @@ class AccountJournal(models.Model):
 
     @api.model
     def _fill_missing_values(self, vals):
-        super()._fill_missing_values(vals)
+        values = super()._fill_missing_values(vals)
 
-        if vals.get('type') != 'purchase':
+        if values.get('type') != 'purchase':
             return
 
-        company = self.env['res.company'].browse(vals['company_id']) if vals.get('company_id') else self.env.company
+        company = self.env['res.company'].browse(values['company_id']) if values.get('company_id') else self.env.company
         if company.country_id.code == "NL":
-            type_control_ids = vals.get('type_control_ids', [])
+            type_control_ids = values.get('type_control_ids', [])
             type_control_ids.append(self.env.ref('account.data_account_type_direct_costs').id)
-            vals['type_control_ids'] = [(6, 0, type_control_ids)]
+            values['type_control_ids'] = [(6, 0, type_control_ids)]
```

Comment:
You should fill a value only if there is a missing value.
You should add something like this `and not values.get('type_control_ids')[0][2]` in the condition to check that there isn't already something filled in.

Reply:
Also, could you change the `(6,0,0)` for an ORM command?